### PR TITLE
Add Interface to Security Component

### PR DESCRIPTION
**Pull request by mostafakhudair**
While Security Class is one of core system classes and it might be replaced by user Security class, it should have it's own interface for more flexibility.

- [x] Securely signed commits
- [x] Component(s) with PHPdocs

### CODE REVIEW

File: system/Security/Exceptions/SecurityException.php [hunk 1]
```diff
@@ -11,9 +11,10 @@
 
 namespace CodeIgniter\Security\Exceptions;
 
+use CodeIgniter\Exceptions\ExceptionInterface;
 use CodeIgniter\Exceptions\FrameworkException;
 
-class SecurityException extends FrameworkException
+class SecurityException extends FrameworkException implements ExceptionInterface
```
Comment (paulbalandan) [inline system/Security/Exceptions/SecurityException.php:17]: I'm sure FrameworkException already implements ExceptionInterface, so there is absolutely no need to reimplement it here. Please revert changes made to this file.

File: system/Security/SecurityInterface.php [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+	 *
+	 * @param RequestInterface $request
+	 *
+	 * @return $this|false
+	 * 
+	 * @throws SecurityException
+	 */
+	public function verify(RequestInterface $request);
+
+	//--------------------------------------------------------------------
```
Comment (paulbalandan) [inline system/Security/SecurityInterface.php:33]: Please remove ALL code separator comments used in this file.

**Review by paulbalandan — CHANGES_REQUESTED** [review]

File: system/Security/SecurityInterface.php (hunk 2 above)
Comment (mostafakhudair) [reply system/Security/SecurityInterface.php:33]: Why? all methods in the framework interfaces are separated by comment line,

Comment (paulbalandan) [reply system/Security/SecurityInterface.php:33]: _Not all_ interfaces, actually.

Up until now, code separator comments serve no real purpose than to delineate perfectly readable code. These just add unnecessary lines of non functional code.

Although these sort of comments are used frequently does not mean it is empirically right when it adds no value itself. We're just being economic here. Besides, there is an automated fix to remove all existing code separators if the need be, but it is just not the priority now.

Comment (lonnieezell) [reply system/Security/SecurityInterface.php:33]: This is a style holdover from CI3 that we're slowly removing as we process files. While it provides a nice visual cue, most decent editors/IDEs today provide a way to do something similar. And if not, the comment block does a good job of it on its own.

**Comment by samsonasik** [comment]
please rebase instead of merge

**Comment by samsonasik** [comment]
I wrote a blog post how to rebase against feature branch https://samsonasik.wordpress.com/2015/09/16/practical-git-4-rebasing-conflicted-task-branch-against-primary-branch/

**Pull request closed without merging**